### PR TITLE
Getting ready for production

### PR DESCRIPTION
**Pull request by axilleas**
Still missing the network app. MariaDB was replaced by PostgreSQL, firewalld was extracted to its own role, nginx config is almost ready, added a Vagrantfile.

- [x] firewalld
- [x] nginx
- [x] vagrant support

satnogs-network django app will be in a different PR.

### CODE REVIEW

File: main.yml [hunk 1]
```diff
@@ -1,9 +1,12 @@
 ---
 - hosts: all
   sudo: yes
+  sudo_user: root
```
Comment (comzeradd) [inline main.yml]: Why root and not a sudo regular user?

**Comment by comzeradd** [comment]
:beer:

File: main.yml [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
   gather_facts: True
 
   roles:
     - common
-    - nginx
-    - mariadb
+    - { role: firewalld, tags: ['firewall'] }
+    - { role: nginx, tags: ['nginx'] }
+    - { role: postgres, tags: ['postgres', 'database'] }
```
Comment (comzeradd) [inline main.yml]: Let's wait till tomorrow's meeting to decide if we want to go mariadb or postgres.

File: roles/nginx/vars/main.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+# vars file for nginx
+
+sitename: "network.satnogs.org"
```
Comment (comzeradd) [inline roles/nginx/vars/main.yml]: Keep in mind that we have two services running on this vm (network, db) on two modes (prod, dev). See [this issue](https://github.com/satnogs/satnogs-org/issues/120).

File: main.yml (hunk 2 above)
Comment (comzeradd) [inline main.yml]: We briefly discussed it and we generally feel more comfortable with MariaDB. So let's revert this and use the existing MariaDB role/tasks.

File: main.yml (hunk 1 above)
Comment (axilleas) [inline main.yml]: Because yolo

File: roles/nginx/vars/main.yml (hunk 3 above)
Comment (axilleas) [inline roles/nginx/vars/main.yml]: ff220c2 fixes that.

File: main.yml (hunk 2 above)
Comment (axilleas) [inline main.yml]: Ack, reverted.

File: main.yml (hunk 1 above)
Comment (axilleas) [inline main.yml]: I found it causing issues with directory permissions when `sudo_user` was defined in a task.

**Comment by comzeradd** [comment]
You should probably squash the last (merge) commit.

**Comment by axilleas** [comment]
Should be good to merge. We'll do the django apps in a separate pr.

**Comment by comzeradd** [comment]
go go go :rocket:

**Pull request merged**